x/btcstaking/keeper/msg_server.go

```diff
@@ -658,19 +711,8 @@ func (ms msgServer) BTCUndelegate(goCtx context.Context, req *types.MsgBTCUndele
 
 		types.EmitEarlyUnbondedEvent(ctx, btcDel.MustGetStakingTxHash().String(), stakerSpendigTxHeader.Height)
 	} else {
-		// stakeSpendingTx is not unbonding tx, first we need to verify whether it
-		// actually spends staking output
-		stakingTxHash, err := chainhash.NewHashFromStr(req.StakingTxHash)
-
-		if err != nil {
-			// panic as we already verified the staking tx hash in the beginning
-			panic(fmt.Errorf("failed to parse staking tx hash from existing delegation with hash %s: %w", req.StakingTxHash, err))
-		}
-
-		if !containsInput(stakeSpendingTx, stakingTxHash, btcDel.StakingOutputIdx) {
-			return nil, types.ErrInvalidBTCUndelegateReq.Wrap("stake spending tx does not spend staking output")
-		}
-
+		// spend staking tx is not the registered unbonding tx, we need to save it in the database
```

Comment:
Do we need an alert for this case?

Reply:
hmm probably, we emit event about with all necessary data to identify transaction so:
1. we can have alert (maybe a warning one)
2. after that manual interverion would be needed to check saved transaction